src/condor_tools/htcondor_cli/annex_create.py

```diff
@@ -688,6 +867,162 @@ def create_annex_token(logger, type):
+            raise ValueError(error_string)
+
+        if cpus is not None and cpus > queue['cores_per_node']:
+            error_string = f"The '{queue_name}' queue is limited to {queue['cores_per_node']} cores per node."
+            raise ValueError(error_string)
+
+        # Don't allow the user to request more than max nodes.
+        mnpj = queue['max_nodes_per_job']
+        if nodes > mnpj:
+            error_string = f"The '{queue_name}' queue is limited to {mnpj} nodes per job.  Use --nodes to set."
+            raise ValueError(error_string)
+
+    # (J) If the queue's allocation type is 'gpu-shared', then specifying
+    #     CPUs or memory is pointless.
+    if queue['allocation_type'] == 'gpu-shared':
+        if cpus is not None or mem_mb is not None:
+            error_string = f"The '{queue_name}' queue always assigns CPUs and RAM in proportiona to the ratio of requested GPUs to total GPUs.  You can't specify CPUs (--cpus) or memory (--mem_mb)."
+            raise ValueError(error_string)
+
+        # Don't allow the user to request more than max nodes.
+        mnpj = queue['max_nodes_per_job']
+        if nodes > mnpj:
+            error_string = f"The '{queue_name}' queue is limited to {mnpj} nodes per job.  Use --nodes to set."
+            raise ValueError(error_string)
+
+    # (E) You must not specify a lifetime longer than the queue's duration.
+    if queue['max_duration'] < lifetime_in_seconds:
+        error_string = f"The '{queue_name}' queue has a maximum duration of {queue['max_duration']} seconds, which is less than the requested lifetime ({lifetime_in_seconds} seconds).  Use --lifetime to set."
+        raise ValueError(error_string)
+
+    # (F) You must not specify an idletime longer than the queue's duration.
+    #     There's an argument that doing so is the natural way to disable
+    #     the idleness check entirely, so let's skip this check for now.
+    # if queue['max_duration'] < idletime_in_seconds:
+    #    error_string = f"The '{queue_name}' queue has a maximum duration of {queue['max_duration']} seconds, which is less than the requeqested lifetime ({idletime_in_seconds} seconds).  Use --idle-time to set."
+    #    raise ValueError(error_string)
+
+    # (G) You must not specify an idletime longer than the lifetime.
+    if lifetime_in_seconds < idletime_in_seconds:
+        error_string = f"You may not specify an idle time (--idle-time) longer than the lifetime (--lifetime)."
+        raise ValueError(error_string)
+
+    # (H) Some systems require an allocation be specified, even if it's
+    #     your only one.
+    if allocation is None and system.allocation_required:
+        error_string = f"The system named '{system.pretty_name}' requires you to specify a project (--project), even if you only have one."
+        raise ValueError(error_string)
+
+    # (I) You must request an appropriate number of GPUs if the queue
```

Comment:
So that if you specify a number of GPUs it has to be correct, but otherwise we'll just use the right number?

That makes sense, I guess.  (I like the idea of making the user be explicit about what their requests mean, although I guess it's also weird to force them to repeat things you already know.

I'll fix this.

Reply:
[Fixed](https://github.com/htcondor/htcondor/pull/943/commits/d60a35f92a40f5bdff6e7eac83490b33cc24e554).